bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioFormat.java

```diff
@@ -238,6 +240,132 @@ public boolean isCompatible(AudioFormat audioFormat) {
+        // Return the first concrete AudioFormat found
+        for (AudioFormat currentAudioFormat : audioFormats) {
+            // Check if currentAudioFormat is abstract
+            if (null == currentAudioFormat.getCodec()) {
+                continue;
+            }
+            if (null == currentAudioFormat.getContainer()) {
+                continue;
+            }
+            if (null == currentAudioFormat.isBigEndian()) {
+                continue;
+            }
+            if (null == currentAudioFormat.getBitDepth()) {
+                continue;
+            }
+            if (null == currentAudioFormat.getBitRate()) {
+                continue;
+            }
+            if (null == currentAudioFormat.getFrequency()) {
+                continue;
+            }
+
+            // Prefer WAVE container
+            if (!currentAudioFormat.getContainer().equals("WAVE")) {
+                continue;
+            }
+
+            // As currentAudioFormat is concrete, use it
+            return currentAudioFormat;
+        }
+
+        // There's no concrete AudioFormat so we must create one
+        for (AudioFormat currentAudioFormat : audioFormats) {
+            // Define AudioFormat to return
+            AudioFormat format = currentAudioFormat;
+
+            // Not all Codecs and containers can be supported
+            if (null == format.getCodec()) {
+                continue;
+            }
+            if (null == format.getContainer()) {
+                continue;
+            }
+
+            // Prefer WAVE container
+            if (!format.getContainer().equals(AudioFormat.CONTAINER_WAVE)) {
+                continue;
+            }
+
+            // If required set BigEndian, BitDepth, BitRate, and Frequency to default values
+            if (null == format.isBigEndian()) {
+                format = new AudioFormat(format.getContainer(), format.getCodec(), new Boolean(true),
+                        format.getBitDepth(), format.getBitRate(), format.getFrequency());
+            }
+            if (null == format.getBitDepth() || null == format.getBitRate() || null == format.getFrequency()) {
+                // Define default values
+                int defaultBitDepth = 16;
+                long defaultFrequency = 16384;
+
+                // Obtain current values
+                Integer bitRate = format.getBitRate();
+                Long frequency = format.getFrequency();
+                Integer bitDepth = format.getBitDepth();
+
+                // These values must be interdependent (bitRate = bitDepth * frequency)
+                if (null == bitRate) {
+                    if (null == bitDepth) {
+                        bitDepth = new Integer(defaultBitDepth);
+                    }
+                    if (null == frequency) {
+                        frequency = new Long(defaultFrequency);
+                    }
+                    bitRate = new Integer(bitDepth.intValue() * frequency.intValue());
```

Comment:
Just to be sure: If frequency is a "long", does it make sense to use only the int range for calculation? I assume that frequency is not higher as the max. int, but as long has been choosen, I want to be sure.